nibabel/streamlines/tests/test_array_sequence.py

```diff
@@ -91,11 +92,20 @@ def test_creating_arraysequence_from_list(self):
                       SEQ_DATA['data'])
 
     def test_creating_arraysequence_from_generator(self):
-        gen = (e for e in SEQ_DATA['data'])
-        check_arr_seq(ArraySequence(gen), SEQ_DATA['data'])
+        gen_1, gen_2 = itertools.tee((e for e in SEQ_DATA['data']))
+        seq = ArraySequence(gen_1)
+        seq_with_buffer = ArraySequence(gen_2, buffer_size=256)
+
+        # Check buffer size effect
+        assert_true(seq_with_buffer.data.shape[0] > seq.data.shape[0])
```

Comment:
Interesting, should we make sure that once creating is done we have the same `._data.shape[0]`?

Reply:
Sorry, I do not understand what you mean here

Reply:
I just meant depending on the buffer_size you chose you might find yourself with a lot of "reserved" allocated space. Let say you have only one streamline in `streamlines` and do `arr_seq = ArraySequence(streamlines, buffer_size=1024)` then your `arr_seq` will be using 1024. 

I thought we might want to trim `arr_seq._data` at the end of the `extend` in order to reduce memory consumption. On the other hand, using the default `buffer_size=4` would only consume up to 4Mb which is rather small.

Reply:
ok, I see, I was surprised too :-).  For sure it will be good to trim the data. We should create a new PR for that, I suppose.